graylog2-web-interface/src/views/components/widgets/FieldSortIcon.jsx

```diff
@@ -0,0 +1,101 @@
+// @flow strict
+import React from 'react';
+import styled, { css, type StyledComponent } from 'styled-components';
+import PropTypes from 'prop-types';
+
+import MessagesWidgetConfig, { defaultSortDirection } from 'views/logic/widgets/MessagesWidgetConfig';
+import Direction, { type DirectionJson } from 'views/logic/aggregationbuilder/Direction';
+import SortConfig from 'views/logic/aggregationbuilder/SortConfig';
+import CustomPropTypes from 'views/components/CustomPropTypes';
+
+import { Icon } from 'components/common';
+
+type Props = {
+  config: MessagesWidgetConfig,
+  fieldName: string,
+  onConfigChange: (MessagesWidgetConfig) => Promise<void>,
+  setLoadingState: (loading: boolean) => void,
+}
+
+const SortIcon: StyledComponent<{fieldSortActive: boolean}, {}, HTMLButtonElement> = styled.button(({ fieldSortActive }) => {
+  const color = fieldSortActive ? '#333' : '#bdbdbd';
+
+  return css`
+    border: 0;
+    background: transparent;
+    color: ${color};
+
+    padding: 5px;
+
+    cursor: pointer;
+  `;
+});
+
+const _newSort = (config, fieldName, currentSortDirectioName) => {
```

Comment:
I've had some inconsistent behaviour when I tried to compare e.g. `Direction.Ascending` with the direction of a widget config sort and decided to use the direction name instead of the class.